### PR TITLE
fix `The document.open function throws an error because document.defaultView returns null` (close #1272)

### LINKED ISSUE #1272: The document.open function throws an error because document.defaultView returns null
Error message: `Uncaught TypeError: Cannot read property 'top' of null`
Find on http://www.virgilio.it/.
Example for reproducing:
```html
<body>
<script>
	var counter = 0;
	var getHtml = function () {
		return '<div>Hello ' + (counter++) + '</div>';
	};
	
	var loadHandler = function (iframe) {
		var win = iframe.contentWindow;
		var doc = iframe.contentDocument;
		
		win.__renderer__ || (win.__renderer__ = true, setTimeout(function () {
			doc.open();
			doc.write(getHtml());
			win.__renderer__ = true;
			doc.close();
		}, 0));
	}
	
	var div = document.createElement('div');
	
	div.innerHTML = '<iframe src="javascript:\'\';" onload="eval(\'(\' + loadHandler' +
					'.toString() + \')(this);\')"></iframe>';
	
	var iframe = div.firstChild;
	
	document.body.appendChild(iframe);
</script>
</body>
```

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @miherlosev @churkin

### CODE REVIEW

File: test/client/fixtures/sandbox/node/document-test.js [hunk 1]
```diff
@@ -494,23 +495,66 @@ if (!browserUtils.isIE) {
...
+            strictEqual(selfRemovingScripts.length, 0);
+        });
+});
 
-    iframeDocument.designMode = 'on';
-    iframeDocument.open();
-    iframeDocument.write('<body style=\"padding: 0; margin: 0; overflow: hidden;\"></body>');
-    iframeDocument.close();
+if (browserUtils.isWebKit) {
+    test('should not throw an error when document loses context window from defaultView property (GH-1272)', function () {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/node/document-test.js]: 'should not throw an error when document.defualtView is null`

File: test/client/fixtures/sandbox/node/document-test.js [hunk 2]
```diff
@@ -494,23 +495,66 @@ if (!browserUtils.isIE) {
...
-    var iframeDocument = iframe.contentDocument;
+            strictEqual(selfRemovingScripts.length, 0);
+        });
+});
 
-    iframeDocument.designMode = 'on';
-    iframeDocument.open();
-    iframeDocument.write('<body style=\"padding: 0; margin: 0; overflow: hidden;\"></body>');
-    iframeDocument.close();
+if (browserUtils.isWebKit) {
```
Comment (miherlosev) [inline test/client/fixtures/sandbox/node/document-test.js]: I guess we can drop this instruction.
Test should pass in all supported browsers.

Comment (LavrovArtem) [inline test/client/fixtures/sandbox/node/document-test.js]:
```javascript
loadEventCount === 2
```
Only in webkit browsers this iframe twice emit the `load` event.

Comment (miherlosev) [inline test/client/fixtures/sandbox/node/document-test.js]: Ok

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
FPR

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
FPR

**Pull request closed without merging**